src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -123,27 +129,145 @@ namespace AppInstaller::Repository
         {
-            return TryGetSourcesFromSetting(settingName).value_or(std::vector<SourceDetails>{});
+            return TryGetSourcesFromSetting(setting, rootName, parse).value_or(std::vector<SourceDetailsInternal>{});
+        }
+
+        // Gets the metadata.
+        std::vector<SourceDetailsInternal> GetMetadata()
+        {
+            return GetSourcesFromSetting(
+                Settings::Streams::SourcesMetadata,
+                s_MetadataYaml_Sources,
+                [&](SourceDetailsInternal& details, const std::string& settingValue, const YAML::Node& source)
+                {
+                    std::string_view name = Settings::Streams::SourcesMetadata.Path;
+                    if (!TryReadScalar(name, settingValue, source, s_MetadataYaml_Source_Name, details.Name)) { return false; }
+                    int64_t lastUpdateInEpoch{};
+                    if (!TryReadScalar(name, settingValue, source, s_MetadataYaml_Source_LastUpdate, lastUpdateInEpoch)) { return false; }
+                    details.LastUpdateTime = Utility::ConvertUnixEpochToSystemClock(lastUpdateInEpoch);
+                    return true;
+                });
+        }
+
+        // Gets the sources from a particular origin.
+        std::vector<SourceDetailsInternal> GetSourcesByOrigin(SourceOrigin origin)
+        {
+            std::vector<SourceDetailsInternal> result;
+
+            switch (origin)
+            {
+            case SourceOrigin::Default:
+            {
+                SourceDetailsInternal details;
+                details.Name = s_Source_WingetCommunityDefault_Name;
+                details.Type = Microsoft::PreIndexedPackageSourceFactory::Type();
+                details.Arg = s_Source_WingetCommunityDefault_Arg;
+                details.Data = s_Source_WingetCommunityDefault_Data;
+                result.emplace_back(std::move(details));
+            }
+                break;
+            case SourceOrigin::User:
+                result = GetSourcesFromSetting(
+                    Settings::Streams::UserSources,
+                    s_SourcesYaml_Sources,
+                    [&](SourceDetailsInternal& details, const std::string& settingValue, const YAML::Node& source)
+                    {
+                        std::string_view name = Settings::Streams::UserSources.Path;
+                        if (!TryReadScalar(name, settingValue, source, s_SourcesYaml_Source_Name, details.Name)) { return false; }
+                        if (!TryReadScalar(name, settingValue, source, s_SourcesYaml_Source_Type, details.Type)) { return false; }
+                        if (!TryReadScalar(name, settingValue, source, s_SourcesYaml_Source_Arg, details.Arg)) { return false; }
+                        if (!TryReadScalar(name, settingValue, source, s_SourcesYaml_Source_Data, details.Data)) { return false; }
+                        int32_t isTombstoneNumber{};

```

Comment:
>int32_t isTombstoneNumber{}; [](start = 24, length = 28)

why is it not a bool in yaml? is it easier this way?

Reply:
Because I didn't realize that there was a bool type in YAML...

Reply:
Changed it, and it is far easier to just use bool.